### PR TITLE
Added Realm Update Functionality

### PR DESCRIPTION
**Pull request by rfoster2576**
## JIRA ID
AC-93457

## Additional Information
Added the ability to update the Realm configuration/values

### CODE REVIEW

**Review by Michael-Valeiras — COMMENTED** [review]
Test comment. Have not reviewed yet.

File: pkg/common/cluster_actions.go [hunk 1]
```diff
@@ -126,6 +127,13 @@ func (i *ClusterActionRunner) CreateRealm(obj *v1alpha1.KeycloakRealm) error {
 	return err
 }
 
+func (i *ClusterActionRunner) UpdateRealm(obj *v1alpha1.KeycloakRealm) error {
+	if i.keycloakClient == nil {
```
Comment (Michael-Valeiras) [inline pkg/common/cluster_actions.go:131]: Should this be a null check on the keycloakClient or the keycloakRealm?

File: pkg/common/cluster_actions.go [hunk 2]
```diff
@@ -367,6 +375,11 @@ type CreateRealmAction struct {
 	Msg string
 }
 
+type UpdateRealmAction struct {
```
Comment (Michael-Valeiras) [inline pkg/common/cluster_actions.go:378]: Question: Are you aware of the difference in usage between this struct and the one below "ConfigureRealmAction" which seems to not follow the pattern of Actions established here?

File: pkg/controller/keycloakrealm/keycloakrealm_reconciler_test.go [hunk 3]
```diff
@@ -57,6 +57,53 @@ func getDummyRealm() *v1alpha1.KeycloakRealm {
...
+			},
+			Realm: &v1alpha1.KeycloakAPIRealm{
+				ID:                        "dummy",
+				Realm:                     "dummy",
+				Enabled:                   true,
+				DisplayName:               "updated",
+				EventsEnabled:             &[]bool{false}[0],
+				AdminEventsEnabled:        &[]bool{false}[0],
+				AdminEventsDetailsEnabled: &[]bool{false}[0],
+				Attributes: map[string]string{
```
Comment (Michael-Valeiras) [inline pkg/controller/keycloakrealm/keycloakrealm_reconciler_test.go:82]: Related to your previous scenario. Does this test cover all updates cases? (addition of value, change in value, deletion of value) and check for all those cases?

**Review by Michael-Valeiras — CHANGES_REQUESTED** [review]
In addition to question and comments left. Have you verified that there is no new auto generated code based on any of the changes made here?

File: pkg/common/cluster_actions.go (hunk 2 above)
Comment (rfoster2576) [reply pkg/common/cluster_actions.go:378]: The Update Realm Struct is used for updating the Realm properties. It looks like the Configure Realm Action is used for Realm overrides, but more specifically for configuring browser redirect.

File: pkg/common/cluster_actions.go (hunk 1 above)
Comment (rfoster2576) [reply pkg/common/cluster_actions.go:131]: It should be a null check on keycloakClient. KeycloakRealm is not a property of the ClusterActionRunner. The keycloakClient in this case is not a Client Resource. Their naming in this case can be confusing

File: pkg/controller/keycloakrealm/keycloakrealm_reconciler_test.go (hunk 3 above)
Comment (rfoster2576) [reply pkg/controller/keycloakrealm/keycloakrealm_reconciler_test.go:82]: It currently tests updating values and adding an attribute

**Review by Michael-Valeiras — APPROVED** [review]
LGTM

**Pull request merged**